dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheHttpClientChannels.java

```diff
@@ -92,9 +96,37 @@ public static CloseableClient createCloseableHttpClient(ClientConfiguration conf
                 !conf.fallbackToCommonNameVerification(), "fallback-to-common-name-verification is not supported");
         Preconditions.checkArgument(!conf.meshProxy().isPresent(), "Mesh proxy is not supported");
 
+        ApacheClientGauges.install(conf.taggedMetricRegistry());
+
         long socketTimeoutMillis =
                 Math.max(conf.readTimeout().toMillis(), conf.writeTimeout().toMillis());
         int connectTimeout = Ints.checkedCast(conf.connectTimeout().toMillis());
+
+        SocketConfig socketConfig = SocketConfig.custom().setSoKeepAlive(true).build();
+        SSLConnectionSocketFactory sslSocketFactory = new SSLConnectionSocketFactory(
+                conf.sslSocketFactory(),
+                new String[] {"TLSv1.2"},
+                conf.enableGcmCipherSuites()
+                        ? jvmSupportedCipherSuites(CipherSuites.allCipherSuites())
+                        : jvmSupportedCipherSuites(CipherSuites.fastCipherSuites()),
+                new DefaultHostnameVerifier());
+
+        PoolingHttpClientConnectionManager connectionManager =
+                new PoolingHttpClientConnectionManager(RegistryBuilder.<ConnectionSocketFactory>create()
+                        .register("http", PlainConnectionSocketFactory.getSocketFactory())
+                        .register("https", sslSocketFactory)
+                        .build());
+
+        ApacheClientGauges.register(connectionManager);
+
+        connectionManager.setDefaultSocketConfig(socketConfig);
+        connectionManager.setMaxTotal(Integer.MAX_VALUE);
+        connectionManager.setDefaultMaxPerRoute(Integer.MAX_VALUE);
+        // Increased from two seconds to twenty-five seconds because we have strong support for retries
+        // and can optimistically avoid expensive connection checks.
+        connectionManager.setValidateAfterInactivity(
+                Ints.checkedCast(Duration.ofSeconds(25).toMillis()));
```

Comment:
I snuck in a behavior change here, I can pull this into a separate PR if you prefer. I dropped it here because this PR adds direct access to the PoolingHttpClientConnectionManager, which is the only to configure `setValidateAfterInactivity`